js/glowingbear.js

```diff
@@ -746,6 +746,31 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         }
     };
 
+    function nickFromBufferline(bufferline) {
+        if (!bufferline || !bufferline.prefix) {
+            return '';
+        }
+        var prefix = bufferline.prefix;
+        var nick="";
+        for (var i=0; i < prefix.length; i++) {
+            nick += prefix[i].text;
+        }
+        return nick;
+    }
+    $rootScope.bufferlineClassifier = function(lines, index) {
```

Comment:
Hm, true, it's passed by reference. That doesn't mean that it's necessary or a good idea. And yes you can do that with `activeBuffer`. Your `lines` variable refers to the same thing as `models.getActiveBuffer().lines`. What this should look like is something like this:
```html
<tr ng-repeat="bufferline in bufferlines" class="bufferline" ng-class="::{'repeated-prefix': !!bufferline.nick && bufferline.nick == bufferlines[$index-1].nick}">
```
or something like that. No need for a bufferlineClassifier function polluting `$rootScope`.